packages/liveblocks-core/src/client.ts

```diff
@@ -600,11 +623,136 @@ export function createClient<U extends BaseUserMeta = DU>(
     getThreadsSince,
   } = createNotificationsApi({
     baseUrl,
-    fetcher: clientOptions.polyfills?.fetch || /* istanbul ignore next */ fetch,
+    fetcher,
     authManager,
     currentUserIdStore,
   });
 
+  async function fetchRoomApi(
+    roomId: string,
+    endpoint: URLSafeString,
+    params?: QueryParams,
+    options?: RequestInit
+  ) {
+    if (!endpoint.startsWith("/v2/c/rooms/")) {
+      raise("Expected a /v2/c/rooms/* endpoint");
+    }
+
+    // TODO: Use the right scope
+    const authValue = await authManager.getAuthValue({
+      requestedScope: "room:read",
+      roomId,
+    });
+
+    const url = urljoin(baseUrl, endpoint, params);
+    return await fetcher(url, {
+      ...options,
+      headers: {
+        ...options?.headers,
+        Authorization: `Bearer ${getAuthBearerHeaderFromAuthValue(authValue)}`,
+      },
+    });
+  }
```

Comment:
@nimeshnayaju This code looks really familiar. Where was this copied from?

Reply:
Yes! It was copied from `room.ts` file. Ideally, I'd like to remove all the fetch related logic from `room.ts` and have them call methods from client instead. Currently, I've only implemented `get(Room)Threads` and `get(Room)ThreadsSince` but haven't updated the respective methods on `room.ts` to use the client method instead.

https://github.com/liveblocks/liveblocks/blob/aca360f891ecbf5d2657e64240cef36b2df23c14/packages/liveblocks-core/src/room.ts#L1520-L1538

Reply:
Perfect, as long as we don't forget to remove the original source and accidentally introduce the duplication here 👍